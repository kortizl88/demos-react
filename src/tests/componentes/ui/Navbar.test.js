import {mount} from 'enzyme'
import { Navbar } from '../../../componentes/ui/Navbar';
import { AuthContext } from '../../../auth/AuthContext';
import {MemoryRouter, Router} from 'react-router-dom';
import types from '../../../types/types';

describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }

    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: true,
            name: "kris"
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value = {contextValue} >
            <MemoryRouter>
                <Router history={historyMock} >
                    <Navbar />
                </Router>
             </MemoryRouter>
        </AuthContext.Provider>
    );


    test('Debe mostrarse el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        expect ( wrapper.find('.text-info').text().trim()).toBe("kris")
        
    })

    test('Debe de llamar el logout y usar el history', () => {
      
        wrapper.find('button').prop("onClick")();

        expect( contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        })
        expect( historyMock.replace ).toHaveBeenCalledWith("/login");
    })
    
    
})
