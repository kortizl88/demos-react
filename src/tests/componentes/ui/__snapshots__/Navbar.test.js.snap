// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> Debe mostrarse el componente correctamente 1`] = `ReactWrapper {}`;
